Added figures 8 and 10 to recipe_bock20jgr.yml

schlunma opened the pull request:
<!--
    Thank you for contributing to our project!

    Please do not delete this text completely, but read the text below and keep
    items that seem relevant. If in doubt, just keep everything and add your
    own text at the top, a reviewer will update the checklist for you.

    While the checklist is intended to be filled in by the technical and scientific
    reviewers, it is the responsibility of the author of the pull request to make
    sure all items on it are properly implemented.
-->

This PR adds the figures 8 and 10 of [Bock et al. (2020)](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2019JD032321) to the corresponding recipe.

Note:
- For both figures, 2 CMIP6 models are missing since the data has been removed from the ESGF nodes.
- For figure 10, I use a 5x5 grid to speed the calculation up. For the original paper, a 2x2 grid has been used.

<!--
    Please describe your changes here, especially focusing on why this PR makes
    ESMValTool better and what problem it solves.

    Before you start, please read our [contribution guidelines](https://docs.esmvaltool.org/en/latest/community/introduction.html).

    Please fill in the GitHub issue that is closed by this pull request, e.g. Closes #1903
-->
Details:
- Closes #2073
- Link to documentation: https://esmvaltool--2074.org.readthedocs.build/en/2074/recipes/recipe_bock20jgr.html

* * *

## Before you get started

<!--
    Please discuss your idea with the development team before getting started,
    to avoid disappointment or unnecessary work later. The way to do this is
    to open a new issue on GitHub.
-->

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValTool/issues) to discuss what you are going to do

## Checklist

It is the responsibility of the author to make sure the PR is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

- [x] [🛠][1] PR has a descriptive title for the [changelog](https://docs.esmvaltool.org/en/latest/community/introduction.html#branches-pull-requests-and-code-review)
- [x] [🛠][1] Code follows the [style guide](https://docs.esmvaltool.org/en/latest/community/introduction.html#code-style)
- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/introduction.html#documentation) is available
- [x] [🛠][1] YAML files pass [`pre-commit`](https://docs.esmvaltool.org/en/latest/community/introduction.html#pre-commit) or [`yamllint`](https://docs.esmvaltool.org/en/latest/community/introduction.html#yaml) checks
- [x] [🛠][1] [Circle/CI tests pass](https://docs.esmvaltool.org/en/latest/community/introduction.html#branches-pull-requests-and-code-review)
- [x] [🛠][1] [Codacy code quality checks pass](https://docs.esmvaltool.org/en/latest/community/introduction.html#branches-pull-requests-and-code-review)
- [x] [🛠][1] [Documentation builds successfully](https://docs.esmvaltool.org/en/latest/community/introduction.html#branches-pull-requests-and-code-review) on readthedocs

### New or updated [recipe/diagnostic](https://docs.esmvaltool.org/en/latest/community/diagnostic.html):

- [x] [🛠][1] [Provenance information](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance) has been added
- [x] [🛠][1] New dependencies are added to the [project requirements](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#additional-dependencies)
- [x] [🧪][2] [Documentation](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#documentation) for the recipe/diagnostic clearly describes what it calculates from a scientific point of view
- [x] [🧪][2] Recipe runs successfully on [`@esmvalbot`](https://docs.esmvaltool.org/en/latest/community/introduction.html#running-tests) or some other machine without modification
- [x] [🧪][2] Figure(s)/data [look as expected](https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review)
- [x] [🧪][2] Code is [well documented](https://docs.esmvaltool.org/en/latest/community/introduction.html#what-should-be-documented) and scientifically sound


***

To help with the number pull requests:

-  🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValTool/pulls) in this repository

<!--
If you need help with any of the items on the checklists above, please do not hesitate to ask by commenting in the issue or pull request.
-->

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

schlunma commented:
@esmvalbot Please run recipe_ecs.yml

schlunma commented:
The remaining Codacy issues will not be fixed (would not improved the code).

schlunma commented:
@esmvalbot Please run recipe_ecs.yml

schlunma commented:
@esmvalbot Please run recipe_cox18nature.yml

axel-lauer commented:
@esmvalbot Please run recipe_bock20jgr.yml

schlunma commented:
I will remove the missing models for figure 8 and 10 and comment out the remaining figures for the test.

schlunma commented:
@esmvalbot Please run recipe_bock20jgr_TEST_DO_NOT_MERGE.yml

schlunma commented:
I guess the bot ran out of memory for figure 10 :confused:

nielsdrost commented:
Hi @schlunma, actually, it ran out of disk space :-(

axel-lauer commented:
I am currently running recipe_bock20jgr.yml on Mistral. Output for figures 1, 2, 3, 4 and 8 looks good. Figure 10 is not finished yet...

nielsdrost commented:
> Hi @schlunma, actually, it ran out of disk space :-(

I'll request a larger disk for the bot...

schlunma commented:
> Hi @schlunma, actually, it ran out of disk space :-(

Yeah, this recipe produces a lot of intermediate files  in the `preproc` dir, the actual output should not be a problem.

axel-lauer reviewed (approved):
I ran recipe_bock20jgr.yml and the output looks fine (I had to remove the CMIP5 model CESM1-CAM5-1-FV2 because of missing data). The output of figure 10 is on a coarser grid (5x5 instead of 2x2 as in the orignal paper) but from a scientific point of view this is fine. This could either be reverted to 2x2 degree or alternatively simply mentioned in the documentation to prevent surprises.

Please do not forget to remove recipe_bock20jgr_TEST_DO_NOT_MERGE.yml before merging.

schlunma commented:
Great, thanks for reviewing Axel! I added a note to the [documentation](https://esmvaltool--2074.org.readthedocs.build/en/2074/recipes/recipe_bock20jgr.html) and removed the test recipe.

LisaBock commented:
Thanks, Manuel, for adding the figures! And thanks, Axel, for reviewing!
